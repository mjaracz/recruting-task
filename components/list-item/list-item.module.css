.ListItem {
  margin-top: 10px;
}