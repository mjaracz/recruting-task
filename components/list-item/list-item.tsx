import { FC, Fragment } from "react";
import { default as MaterialListItem } from "@mui/material/ListItem";
import ListItemAvatar from "@mui/material/ListItemAvatar";
import ListItemText from "@mui/material/ListItemText";
import Avatar from "@mui/material/Avatar";
import Typography from "@mui/material/Typography";
import Divider from "@mui/material/Divider";
import styles from "./list-item.module.css";

interface Props {
  avatarUrl: string;
  username: string;
}

export const ListItem: FC<Props> = ({ avatarUrl, username }) => {
  return (
    <>
      <MaterialListItem className={styles.ListItem} alignItems="flex-start">
        <ListItemAvatar>
          <Avatar alt={username} src={avatarUrl} />
        </ListItemAvatar>
        <ListItemText
          primary={
            <Fragment>
              <Typography
                component="span"
                variant="body2"
                sx={{ color: "text.primary", display: "inline" }}
              >
                {username}
              </Typography>
            </Fragment>
          }
        />
      </MaterialListItem>
      <Divider variant="inset" component="li" />
    </>
  );
};
