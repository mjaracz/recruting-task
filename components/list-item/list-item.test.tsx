import { render } from "@testing-library/react"
import { ListItem } from "./list-item"

describe('<ListItem />', () => {
  test('should correct match the snapshot', () => {
    const { baseElement } = render(<ListItem avatarUrl="123" username="mock user" />)

    expect(baseElement).toMatchSnapshot();
  })
})