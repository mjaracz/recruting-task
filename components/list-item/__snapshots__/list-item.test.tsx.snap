// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should correct match the snapshot 1`] = `
<body>
  <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart ListItem css-y0goe-MuiListItem-root"
    >
      <div
        class="MuiListItemAvatar-root MuiListItemAvatar-alignItemsFlexStart css-1hqvr79-MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
        >
          <img
            alt="mock user"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="123"
          />
        </div>
      </div>
      <div
        class="MuiListItemText-root css-1y2w5y4-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1edfpdg-MuiTypography-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 css-6osl5i-MuiTypography-root"
          >
            mock user
          </span>
        </span>
      </div>
    </li>
    <li
      aria-orientation="horizontal"
      class="MuiDivider-root MuiDivider-inset css-qnmhp7-MuiDivider-root"
      role="separator"
    />
  </div>
</body>
`;
